astropy/coordinates/distances.py

```diff
@@ -100,11 +100,15 @@ def __new__(cls, value=None, unit=None, z=None, cosmology=None,
                 distmod=None, parallax=None, dtype=None, copy=True, order=None,
                 subok=False, ndmin=0, allow_negative=False):
 
-        if z is not None:
-            if value is not None or distmod is not None:
-                raise ValueError('Should given only one of `value`, `z` '
-                                 'or `distmod` in Distance constructor.')
+        n_not_none = sum(x is not None for x in [value, z, distmod, parallax])
+        if n_not_none == 0:
+            raise ValueError('None of `value`, `z`, `distmod`, or `parallax` '
```

Comment:
capitalization of this error message, as you say in https://github.com/astropy/astropy/pull/12531#issuecomment-982535620.